osquery/tables/system/hash.cpp

```diff
@@ -108,7 +108,8 @@ MultiHashes hashMultiFromFile(int mask, const std::string& path) {
                           hash.second->update(&buffer[0], size);
                         }
                       }
-                    }));
+                    }),
+                    true);
```

Comment:
How did you find this?

Reply:
This was just looking at our implementation (for Windows, I verified that it _should_ be alright on Linux through tests and verifying `O_NONBLOCK` wasn't going to have issues on linux), for where we make the Windows Async file reads - as I knew this was something we had implemented, and found that we set "open blocking" to `false` by default. On Windows this causes us to use the async file reads, which can happen out of band of the main thread of operation, and occasionally your data can be delivered late. Setting this to `true` ensures we don't open the file with asynchronous reading turned on on Windows.

Reply:
I’m worried that user controlled sockets can DoS the daemon with blocking opens.